### PR TITLE
Add easyhttpcpp/2.1.0 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **easyhttpcpp/2.1.0**

Depends on #1536

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@danimtb The CI job complains that Poco/1.10.1 is not available. As we know, we have that package.

**Comment by madebr** [comment]
@uilianries CI complains because I used `openssl/1.1.1g` as a requirement instead of `openssl/1.0.2t`.
See #1536

**Comment by uilianries** [comment]
@madebr Now it makes sense!

**Comment by uilianries** [comment]
@madebr fixed, but now there is a new problem, it complains about c++11 support.

File: recipes/easyhttpcpp/all/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(cmake_wrapper)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_subdirectory("source_subfolder")
```
Comment (madebr) [inline recipes/easyhttpcpp/all/CMakeLists.txt:7]:
```suggestion
conan_basic_setup()

if(NOT CMAKE_CXX_STANDARD)
    set(CMAKE_CXX_STANDARD 11)
endif()

add_subdirectory("source_subfolder")
```
@uilianries  Would this be *correct*?

Comment (uilianries) [reply recipes/easyhttpcpp/all/CMakeLists.txt:7]: Yes, few other recipes follow this solution.

Comment (madebr) [reply recipes/easyhttpcpp/all/CMakeLists.txt]: Thanks

**Comment by madebr** [comment]
Wow, not so easy after all.
Please review @SSE4 @uilianries @danimtb

File: recipes/easyhttpcpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        if self.settings.build_type == "Debug":
+            if self.settings.os == "Windows":
+                if not self.options.shared:
+                    libsuffix += "md"
+            libsuffix += "d"
+        self.cpp_info.libs = ["easyhttp{}".format(libsuffix)]
+        if self.settings.os == "Windows" and self.options.shared:
+            self.cpp_info.defines.append("EASYHTTPCPP_DLL")
+        self.cpp_info.name["cmake_find_package"] = "easyhttpcppeasyhttp"
+        self.cpp_info.name["cmake_find_package_multi"] = "easyhttpcppeasyhttp"
```
Comment (madebr) [inline recipes/easyhttpcpp/all/conanfile.py:96]:
```suggestion
        self.cpp_info.names["cmake_find_package"] = "easyhttpcppeasyhttp"
        self.cpp_info.names["cmake_find_package_multi"] = "easyhttpcppeasyhttp"
```

File: recipes/easyhttpcpp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        libsuffix = ""
+        if self.settings.build_type == "Debug":
+            if self.settings.os == "Windows":
+                if not self.options.shared:
+                    libsuffix += "md"
+            libsuffix += "d"
+        self.cpp_info.libs = ["easyhttp{}".format(libsuffix)]
+        if self.settings.os == "Windows" and self.options.shared:
+            self.cpp_info.defines.append("EASYHTTPCPP_DLL")
+        self.cpp_info.names["cmake_find_package"] = "easyhttpcppeasyhttp"
```
Comment (prince-chrismc) [inline recipes/easyhttpcpp/all/conanfile.py:95]:
```suggestion
        # https://github.com/sony/easyhttpcpp/blob/2.1.0/cmake/easyhttpcppeasyhttpConfig.cmake.in
        self.cpp_info.names["cmake_find_package"] = "easyhttpcppeasyhttp" 
```

I assumed it was a copy paste error until I checked. It's just so wrong.

**Comment by madebr** [comment]
@uilianries 
I think your change request was addressed. Can you re-review and unblock?

**Pull request merged**